code/game/machinery/contraband_detector.dm

```diff
@@ -0,0 +1,215 @@
+	var/mob_cooldown = 0
+
+	wires = /datum/wires/contraband_detector
+	var/identifier_wire_pulsed_until = 0	// world time value, since it's only checked in 1 place we don't remove it when time has passed (more optimal than checking every tick)
+
+/obj/machinery/contraband_detector/New()
+	announce = new /obj/item/device/radio/intercom(src)
+	announce.internal_channels = list(num2text(SEC_LCZ_FREQ) = list(ACCESS_SECURITY_LVL2))
+	. = ..()
+	if(anchored)
+		RegisterSignal(loc, COMSIG_ENTERED, .proc/detect_contraband)
+
+/obj/machinery/contraband_detector/Initialize()
+	. = ..()
+	announce.internal_channels = list(num2text(SEC_LCZ_FREQ) = list(ACCESS_SECURITY_LVL2))
+	announce.set_frequency(SEC_LCZ_FREQ)
+
+/obj/machinery/contraband_detector/set_broken(new_state)
+	. = ..()
+	if(new_state)
+		announce_tampering()
+
+/obj/machinery/contraband_detector/ex_act(severity)
+	. = ..()
+	if(prob(100 - (severity * 20)))
+		set_broken(TRUE)
+
+/obj/machinery/contraband_detector/proc/detect_contraband(turf/T, atom/movable/A)	// T isn't used, but it's passed by the signal as first argument so don't remove it
+	if(!wires.is_cut(WIRE_CONTRADETECT_IDENTIFIER) && (((identifier_wire_pulsed_until > world.time) && identifier_wire_pulsed_until != 0) || (istype(A) && A.has_contraband())))
+		if(ismob(A))
+			if(mob_cooldown < world.time)
+				mob_cooldown = world.time + 1 SECOND
+			else
+				return
+		else
+			if(obj_cooldown < world.time)
+				obj_cooldown = world.time + 2 SECONDS
+			else
+				return
+		flick("detected", src)
+		visible_message(SPAN_WARNING("\The [src] detects contraband on \the [A.name]!"))
+		if(!wires.is_cut(WIRE_CONTRADETECT_ALARM))
+			var/area/AR = get_area(src)
+			announce.autosay("Contraband detected on \the [A] at \the [AR.name]!", "Contraband Detection System")
+
+/obj/machinery/contraband_detector/proc/announce_tampering()
+	var/area/AR = get_area(src)
+	announce.autosay("ALERT! Tampering detected on \the [src] at [AR.name]!", "Contraband Detection System")
+
+/obj/machinery/contraband_detector/powered()
+	if(wires.is_cut(WIRE_MAIN_POWER1) || wires.is_cut(WIRE_MAIN_POWER2))
+		return FALSE
+	return ..()
+
+/obj/machinery/contraband_detector/on_update_icon()
+	if(stat & BROKEN)
+		icon_state = "broken"
+	else if(stat & NOPOWER)
+		icon_state = "unpowered"
+	else
+		icon_state = "powered"
+
+/obj/machinery/contraband_detector/attackby(obj/item/I, mob/user)
+	if(isScrewdriver(I))
+		if(id_locked && !panel_open)	// you can close it when ID locked, just not open it
+			to_chat(user, SPAN_NOTICE("\The [src]'s panel is locked!"))
+		else
+			panel_open = !panel_open
+			user.visible_message(SPAN_WARNING("[user] screws the camera's panel [panel_open ? "open" : "closed"]!"),
+			SPAN_NOTICE("You screw the camera's panel [panel_open ? "open" : "closed"]."))
+			playsound(src.loc, 'sound/items/Screwdriver.ogg', 50, 1)
+	if((isWirecutter(I) || isMultitool(I)) && panel_open)
+		return wires.Interact(user)
```

Comment:
Same with this, dont override just do any special checks and use obj/machinery's unarmedAttack override